pkg/apis/hive/v1/agent/installstrategy.go

```diff
@@ -0,0 +1,84 @@
+package agent
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InstallStrategy is the install strategy configuration for provisioning a cluster with the
+// Agent based assisted installer.
+type InstallStrategy struct {
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking Networking `json:"networking"`
+
+	// TODO: consider moving higher into ClusterDeployment in future?
+
+	// SSHPublicKey will be added to all cluster hosts for use in debugging.
+	// +optional
+	SSHPublicKey string `json:"sshPublicKey,omitempty"`
+
+	// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+	ProvisionRequirements ProvisionRequirements `json:"provisionRequirements"`
+}
+
+// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+type ProvisionRequirements struct {
+
+	// ControlPlaneAgents is the number of matching approved and ready Agents with the control plane role
+	// required to launch the install. Must be either 1 or 3.
+	ControlPlaneAgents int `json:"controlPlaneAgents"`
+
+	// WorkerAgents is the minimum number of matching approved and ready Agents with the worker role
+	// required to launch the install.
+	// +optional
+	WorkerAgents int `json:"workerAgents,omitempty"`
+
+	// AgentSelector is a label selector used for associating relevant custom resources with this cluster.
+	// (Agent, BareMetalHost, etc)
+	AgentSelector metav1.LabelSelector `json:"agentSelector"`
+}
+
+// Networking defines the pod network provider in the cluster.
+type Networking struct {
+	// MachineNetwork is the list of IP address pools for machines.
+	// This field replaces MachineCIDR, and if set MachineCIDR must
+	// be empty or match the first entry in the list.
+	// Default is 10.0.0.0/16 for all platforms other than libvirt.
+	// For libvirt, the default is 192.168.126.0/24.
```

Comment:
i think this needs rewording.

Reply:
I originally copied out of installer directly, but those libvirt mentions do seem out of place here. Removing.

Reply:
Ah and removing the mention of the legacy field we didn't carry.